src/main/java/seedu/address/logic/commands/findcommand/FindMentorCommand.java

```diff
@@ -64,4 +72,20 @@ public CommandResult execute(Model model) {
         model.updateHistory(this);
         return new CommandResult(MESSAGE_SUCCESS, CommandType.M);
     }
+
+    @Override
+    public boolean equals(Object otherFindCommand) {
+        if (otherFindCommand == this) {
+            return true;
+        }
+
+        if (!(otherFindCommand instanceof FindMentorCommand)) {
+            return false;
+        }
+
+        return name.equals(((FindMentorCommand) otherFindCommand).name)
+                && email.equals(((FindMentorCommand) otherFindCommand).email)
+                && phone.equals(((FindMentorCommand) otherFindCommand).phone)
+                && organization.equals(((FindMentorCommand) otherFindCommand).organization);
```

Comment:
Maybe you can consider comparing specialisation too, don't think it would make much of a difference hahaha 😊

Reply:
Specialization isnt in command, only the entity